germlineVC.nf

```diff
@@ -91,52 +91,26 @@ if (params.verbose) bamFiles = bamFiles.view {
   Files : [${it[3].fileName}, ${it[4].fileName}]"
 }
 
-// assume input is recalibrated, ignore explicitBqsrNeeded
-(recalibratedBam, recalTables) = bamFiles.into(2)
-
-recalTables = recalTables.map{ it + [null] } // null recalibration table means: do not use --BQSR
-
-recalTables = recalTables.map { [it[0]] + it[2..-1] } // remove status
-
-if (params.verbose) recalibratedBam = recalibratedBam.view {
-  "Recalibrated BAM for variant Calling:\n\
-  ID    : ${it[0]}\tStatus: ${it[1]}\tSample: ${it[2]}\n\
-  Files : [${it[3].fileName}, ${it[4].fileName}]"
-}
-
 // Here we have a recalibrated bam set, but we need to separate the bam files based on patient status.
-// The sample tsv config file which is formatted like: "subject status sample lane fastq1 fastq2"
-// cf fastqFiles channel, I decided just to add _status to the sample name to have less changes to do.
-// And so I'm sorting the channel if the sample match _0, then it's a normal sample, otherwise tumor.
-// Then combine normal and tumor to get each possibilities
-// ie. normal vs tumor1, normal vs tumor2, normal vs tumor3
-// then copy this channel into channels for each variant calling
-// I guess it will still work even if we have multiple normal samples
+// The sample tsv config file which is formatted like: "idPatient status idSample bamFile baiFile"
 
 // separate recalibrateBams by status
-bamsNormal = Channel.create()
-bamsTumor = Channel.create()
-
-recalibratedBam
-  .choice(bamsTumor, bamsNormal) {it[1] == 0 ? 1 : 0}
+bamsToKeep = Channel.create()
+bamsOthers = Channel.create()
 
-// Ascat, Strelka Germline & Manta Germline SV
-bamsForAscat = Channel.create()
-bamsForSingleManta = Channel.create()
-bamsForSingleStrelka = Channel.create()
+bamFiles
+  .choice(bamsOthers, bamsToKeep) {it[1] == 0 ? 1 : 0}
```

Comment:
How does this work? I Have no idea what's going on here...

Reply:
If the status of a set is 0 -> bamsToKeep, else -> bamsOthers
I know that this choice operator can be confusing, especially with no comments.
I'll had some ;-)

Reply:
Ah, I see. You're checking the status field in the tsv. That's the part I was missing.

Reply:
Just added comments, I hope it'll be easier to understand

Reply:
thank you indeed :)

Reply:
Not doing that anymore...
it's now more efficient, and easier to understand I think